tweaks-qt/gen-layout-list

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+filename="/usr/share/X11/xkb/rules/evdev.lst"
+in_layouts=f
+
+cat << EOF
+#pragma once
+
+// Auto-generated based on "$filename"
+
+struct layout
+{
+    const char *code;
+    const char *description;
+};
+
+static struct layout evdev_lst_layouts[] = {
+EOF
+
+while IFS= read -r line; do
+	[ "$line" = "! layout" ] && { in_layouts=t; continue; }
+	[ "$line" = "" ] && in_layouts=f
+	[ "$in_layouts" = "f" ] && continue
+
+	printf '    { "'
+	printf '%b' "$line" | awk '{printf "%s", $1}'
+	printf '", "'
+	printf '%b' "$line" | awk '{for (i=2; i<=NF; i++) printf "%s", $i}'
+	printf '" },\n'
+done < "$filename"
+
+printf '};\n'
```

Comment:
I haven't got yet how this works/should be used, but I think that maybe CMake could manage instead?

Reply:
I did call the gen-script from CMakeLists.txt to start with (by passing the path to `evdev.lst`), but decided to commit the `evdev-lst-layouts.h` file instead to avoid people raising issue when they can't build because `evdev.lst` lives somewhere else on their system.

I don't actually know if parsing `evdev.lst` is even the right thing to do here. There might be better ways to get the layouts from `xkbcommon` or something, but shall we just merge as is until we can either validate that this is a sensible approach - or come up with a better one.